src/keyworddocs.jl

```diff
@@ -0,0 +1,46 @@
+  * `transpose::Bool`: read a csv file "transposed", i.e. each column is parsed as a row
+  * `comment::String`: string that will cause rows that begin with it to be skipped while parsing. Note that if a row number header or `datarow` and `comment` are provided, the header/data row will be the first non-commented/non-empty row _after_ the row number, meaning if the provided row number is a commented row, the header/data row will actually be the next non-commented row.
+  * `ignoreemptyrows::Bool=true`: whether empty rows in a file should be ignored (if `false`, each column will be assigned `missing` for that empty row)
+  * `select`: an `AbstractVector` of `Integer`, `Symbol`, `String`, or `Bool`, or a "selector" function of the form `(i, name) -> keep::Bool`; only columns in the collection or for which the selector function returns `true` will be parsed and accessible in the resulting `CSV.File`. Invalid values in `select` are ignored.
+  * `drop`: inverse of `select`; an `AbstractVector` of `Integer`, `Symbol`, `String`, or `Bool`, or a "drop" function of the form `(i, name) -> drop::Bool`; columns in the collection or for which the drop function returns `true` will ignored in the resulting `CSV.File`. Invalid values in `drop` are ignored.
+  * `limit`: an `Integer` to indicate a limited number of rows to parse in a csv file; use in combination with `skipto` to read a specific, contiguous chunk within a file; note for large files when multiple threads are used for parsing, the `limit` argument may not result in an exact # of rows parsed; use `threaded=false` to ensure an exact limit if necessary
+  * `threaded::Bool`: [not applicable to `CSV.Rows`] whether parsing should utilize multiple threads; by default threads are used on large enough files (more than 5_000 cells), but isn't allowed when `transpose=true`; only available in Julia 1.3+
+  * `ntasks::Integer=Threads.nthreads()`: [not applicable to `CSV.Rows`] for multithreaded parsed files, this controls the number of tasks spawned to read a file in concurrent chunks; defaults to the # of threads Julia was started with (i.e. `JULIA_NUM_THREADS` environment variable or `julia -t N`)
+  * `rows_to_check::Integer=$DEFAULT_ROWS_TO_CHECK`: [not applicable to `CSV.Rows`] a multithreaded parsed file will be split up into `ntasks` # of equal chunks; `rows_to_check` controls the # of rows are checked to ensure parsing correctly found valid rows; for certain files with very large quoted text fields, `lines_to_check` may need to be higher (10, 30, etc.) to ensure parsing correctly finds these rows
+
+## Parsing options:
+
+  * `missingstring`: either a `String`, or `Vector{String}` to use as sentinel values that will be parsed as `missing`; by default, only an empty field (two consecutive delimiters) is considered `missing`
```

Comment:
is `String` or `AbstractString` required here?

Reply:
Currently, Parsers.jl requires `String` strictly; do you think that's overly restrictive? I've never heard anyone complain about it, but maybe once there are more exotic string types out there it will be a bigger deal. We could loosen up the Parsers.jl input allowance and convert to String explicitly.

Reply:
No - I think it is OK for now - just wanted to make sure.

However, the point is that `SubString{String}` is not exotic and often encountered as many functions in Julia Base return it.
Also after CSV.jl release some of the exotic string types will become quite popular. I think we should blame @quinnj for possible exotic string type proliferation 😁.

And seriously - I think it should be easy enough to just cast `AbstractString` to `String` in pre-processing of kwargs, but it can be added later.